crates/errors/src/error_codes/META0016.md

```diff
@@ -0,0 +1,12 @@
+## META0016
+
+Cannot update the provided deployment with the discovered metadata, because the new metadata is insufficiently similar to the old.
+
+When updating a deployment, make sure that:
+
+* All services have the same type as they did in the previous deployment.
+* All services contain at least all the handlers that they did in the previous deployment.
+* The updated deployment contains at least all the services that it previously did.
+* The updated deployment has exactly the same supported protocol versions.
```

Comment:
```suggestion
* The updated deployment has exactly the same supported protocol versions, which usually SDKs guarantee across patch releases of the same minor.
```

Or something like that. just rephrase this a bit to correlate it with SDK versions, as _supported protocol versions_ is not something we expose too much to people.